polars/polars-time/src/windows/window.rs

```diff
@@ -247,17 +261,42 @@ impl Iterator for BoundsIter {
         if self.bi.start < self.boundary.stop {
             let out = self.bi;
             match self.tu {
+                // TODO remove unwrap once time zone is respected
```

Comment:
Do we already have an issue for this? That means we validate timezone upon entry in the API right? I like that idea as it removes a lot of assertions downstream.

Reply:
There's https://github.com/pola-rs/polars/issues/7434

The time zone itself is validated upon entry (e.g. here it's validated in the entry to `date_range_impl`) - the reason for the unwrap isn't that the time zone might not be valid, but that adding a non-fixed offset might result in an ambiguous or non-existent date

For example, if you were to start on `'2021-10-07T01:00 US/Central'` and to add a month, you would end up on `'2021-11-07T01:00 US/Central'` - problem is, is that `'2021-11-07T01:00 CDT'` or `'2021-11-07T01:00 CST'`? Until https://github.com/pola-rs/polars/issues/6672 is addressed, I think the only option would be to error

Reply:
Right, thanks for the explanation. :+1: